paasta_tools/metrics/metastatus_lib.py

```diff
@@ -568,32 +568,41 @@ def run_healthchecks_with_param(param, healthcheck_functions, format_options={})
     return [healthcheck(param, **format_options) for healthcheck in healthcheck_functions]
 
 
-def assert_marathon_apps(client):
-    num_apps = len(client.list_apps())
-    if num_apps < 1:
+def assert_marathon_apps(clients):
+    num_apps = [len(c.list_apps()) for c in clients]
```

Comment:
Do you know if this is tolerant of a shard being down or not?
We could get all sorts of errors from clients, this kind assumes that they will always return a 0.

Can you add error handling... somewhere so that if a shard is down we can return a non-healthy result but at least move on?

Reply:
It a shard is down, it should fail at line 83 of paasta_tools/paasta_metastatus.py

Reply:
In short, this is tolerant of a shard being down because it fails and exits with 2 long before this assertion. This is similar to the way it works now: exit with 2 if either mesos or marathon is down. 

I couldn't come up with a good solution to acknowledge the failure of one or more shards and assert heathy ones.